xarray/core/common.py

```diff
@@ -193,6 +193,11 @@ def __dir__(self):
             if isinstance(item, basestring)]
         return sorted(set(dir(type(self)) + extra_attrs))
 
+    def _ipython_key_completions_(self):
+        item_lists = [item for sublist in self._item_sources for item
+                      in sublist if isinstance(item, basestring)]
```

Comment:
nit: can you reformat this to not line break between "for" and "in", e.g.,
```
        item_lists = [item for sublist in self._item_sources
                      for item in sublist if isinstance(item, basestring)]
```

Actually, additional line breaks might be even better:
```
        item_lists = [item
                      for sublist in self._item_sources
                      for item in sublist
                      if isinstance(item, basestring)]
```